File: 452. Minimum Number of Arrows to Burst Balloons/main.py
from typing import List


class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort()

        index = 0

        for point in points:
            if points[index][1] < point[0]:
                index += 1
                points[index][0] = point[0]
                points[index][1] = point[1]
            else:
                points[index][0] = max(points[index][0], point[0])
                points[index][1] = min(points[index][1], point[1])

        return index + 1


class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])

        count = 1
        preEnd = points[0][1]

        for point in points:
            if point[0] > preEnd:
                count += 1
                preEnd = point[1]

        return count

File: 452. Minimum Number of Arrows to Burst Balloons/main.rs
struct Solution;

impl Solution {
    pub fn find_min_arrow_shots(points: Vec<Vec<i32>>) -> i32 {
        let mut points = points;
        points.sort();

        let mut index = 0;

        for i in 0..points.len() {
            if points[index][1] < points[i][0] {
                index += 1;
                points[index][0] = points[i][0];
                points[index][1] = points[i][1];
            } else {
                points[index][0] = points[index][0].max(points[i][0]);
                points[index][1] = points[index][1].min(points[i][1]);
            }
        }

        index as i32 + 1
    }
}

impl Solution {
    pub fn find_min_arrow_shots(points: Vec<Vec<i32>>) -> i32 {
        let mut points = points;
        points.sort();
    }
}
